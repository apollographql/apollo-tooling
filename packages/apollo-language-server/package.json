{
  "name": "apollo-language-server",
  "description": "A language server for Apollo GraphQL projects",
  "version": "1.26.9",
  "author": "Apollo GraphQL <packages@apollographql.com>",
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/apollographql/apollo-tooling.git"
  },
  "homepage": "https://github.com/apollographql/apollo-tooling",
  "bugs": "https://github.com/apollographql/apollo-tooling/issues",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "engines": {
    "node": ">=8",
    "npm": ">=6"
  },
  "dependencies": {
    "@apollo/federation": "0.27.0",
    "@apollographql/apollo-tools": "0.5.4",
    "@apollographql/graphql-language-service-interface": "^2.0.2",
    "@endemolshinegroup/cosmiconfig-typescript-loader": "^3.0.2",
    "apollo-datasource": "^3.0.0",
    "apollo-env": "^0.10.2",
    "apollo-graphql": "^0.9.7",
    "apollo-link": "^1.2.3",
    "apollo-link-context": "^1.0.9",
    "apollo-link-error": "^1.1.1",
    "apollo-link-http": "^1.5.5",
    "apollo-server-errors": "^2.0.2",
    "await-to-js": "^3.0.0",
    "core-js": "^3.0.1",
    "cosmiconfig": "^7.0.1",
    "dotenv": "^16.0.0",
    "glob": "^8.0.0",
    "graphql": "14.0.2 - 14.2.0 || ^14.3.1 || ^15.0.0",
    "graphql-tag": "^2.10.1",
    "lodash.debounce": "^4.0.8",
    "lodash.merge": "^4.6.1",
    "minimatch": "^5.0.0",
    "vscode-languageserver": "^7.0.0",
    "vscode-languageserver-textdocument": "^1.0.4",
    "vscode-uri": "1.0.6"
  },
  "jest": {
    "preset": "ts-jest",
    "transformIgnorePatterns": [
      "/node_modules/"
    ],
    "setupFiles": [
      "apollo-env"
    ],
    "testEnvironment": "node",
    "testMatch": [
      "**/__tests__/*.(js|ts)"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/lib/",
      "<rootDir>/test/fixtures/",
      "<rootDir>/test/test-utils"
    ],
    "moduleFileExtensions": [
      "ts",
      "js"
    ],
    "globals": {
      "ts-jest": {
        "tsconfig": "<rootDir>/tsconfig.test.json",
        "diagnostics": false
      }
    }
  }
}
