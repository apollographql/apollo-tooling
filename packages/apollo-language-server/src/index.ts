// Exports for consuming APIs

export { getValidationErrors } from "./errors/validation";
export { ToolError } from "./errors/logger";
export { LoadingHandler } from "./loadingHandler";

// projects
export { GraphQLProject } from "./project/base";
export { isClientProject, GraphQLClientProject } from "./project/client";
export { isServiceProject, GraphQLServiceProject } from "./project/service";

// GraphQLSchemaProvider
export {
  GraphQLSchemaProvider,
  schemaProviderFromConfig,
} from "./providers/schema";

// Engine
export * from "./engine";

// Config
export * from "./config";

// Generated types
import * as graphqlTypes from "./graphqlTypes";
export { graphqlTypes };

// GraphQL utilities
export { Debug, withTypenameFieldAddedWhereNeeded } from "./utilities";
