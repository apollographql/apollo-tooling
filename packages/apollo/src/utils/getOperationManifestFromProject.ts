import { GraphQLClientProject } from "apollo-language-server";
import { operationHash, operationRegistrySignature } from "apollo-graphql";

export interface ManifestEntry {
  signature: string;
  document: string;
  metadata: {
    engineSignature: string;
  };
}

export function getOperationManifestFromProject(
  project: GraphQLClientProject,
  options: { preserveStringAndNumericLiterals: boolean } = {
    preserveStringAndNumericLiterals: false,
  }
): ManifestEntry[] {
  const manifest = Object.entries(
    project.mergedOperationsAndFragmentsForService
  ).map(([operationName, operationAST]) => {
    const printed = operationRegistrySignature(operationAST, operationName, {
      preserveStringAndNumericLiterals:
        options.preserveStringAndNumericLiterals,
    });

    return {
      signature: operationHash(printed),
      document: printed,
      // TODO: unused. Remove or repurpose this field altogether with op. registry 2.0 work.
      // For now, this field is non-nullable on the input type.
      metadata: {
        engineSignature: "",
      },
    };
  });

  return manifest;
}
