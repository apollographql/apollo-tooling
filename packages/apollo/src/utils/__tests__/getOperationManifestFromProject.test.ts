import { getOperationManifestFromProject } from "../getOperationManifestFromProject";
import { GraphQLClientProject } from "apollo-language-server";

const mergedOperationsAndFragmentsForService = require("./fixtures/mockOperations.json");

describe("getOperationManifestFromProject", () => {
  const mockProject = {
    mergedOperationsAndFragmentsForService,
  } as GraphQLClientProject;

  it("builds an operation manifest", () => {
    // XXX
    // This is mostly a sanity check, and should prevent someone from unintentionally changing the way
    // that operations are normalized.
    expect(getOperationManifestFromProject(mockProject)).toMatchSnapshot();
  });
});
