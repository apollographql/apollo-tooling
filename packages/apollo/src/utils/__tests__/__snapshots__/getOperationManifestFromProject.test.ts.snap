// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getOperationManifestFromProject builds an operation manifest 1`] = `
Array [
  Object {
    "document": "query SchemaTagsAndFieldStats($id:ID!){service(id:$id){__typename schemaTags{__typename tag}stats(from:\\"\\",to:\\"\\"){__typename fieldStats{__typename groupBy{__typename field}metrics{__typename fieldHistogram{__typename durationMs(percentile:0)}}}}}}",
    "metadata": Object {
      "engineSignature": "",
    },
    "signature": "135e8314de0c2f23f4a2be87b20e59f30ecf2cbcdfad6676a46aad21d29cdb5b",
  },
  Object {
    "document": "mutation CheckSchema($frontend:String,$gitContext:GitContextInput,$historicParameters:HistoricQueryParameters,$id:ID!,$schema:IntrospectionSchemaInput!,$tag:String){service(id:$id){__typename checkSchema(baseSchemaTag:$tag frontend:$frontend gitContext:$gitContext historicParameters:$historicParameters proposedSchema:$schema){__typename diffToPrevious{__typename changes{__typename code description type}type validationConfig{__typename from queryCountThreshold queryCountThresholdPercentage to}}targetUrl}}}",
    "metadata": Object {
      "engineSignature": "",
    },
    "signature": "456e83fbaa97cf470a489ccdbcd2db93c1df84ef39109248f93cabd36ce236f2",
  },
  Object {
    "document": "mutation RegisterOperations($clientIdentity:RegisteredClientIdentityInput!,$id:ID!,$operations:[RegisteredOperationInput!]!){service(id:$id){__typename registerOperations(clientIdentity:$clientIdentity,operations:$operations)}}",
    "metadata": Object {
      "engineSignature": "",
    },
    "signature": "1428ab44b0f2f4ea25bc28c08b5a7235cb96a2d535d1a15e0edd5f0695a4903f",
  },
  Object {
    "document": "mutation UploadSchema($gitContext:GitContextInput,$id:ID!,$schema:IntrospectionSchemaInput!,$tag:String!){service(id:$id){__typename uploadSchema(gitContext:$gitContext,schema:$schema,tag:$tag){__typename code message success tag{__typename schema{__typename hash}tag}}}}",
    "metadata": Object {
      "engineSignature": "",
    },
    "signature": "a2497bb8c32f97870dfc58823b1377f6339c3e5ba53010b2a6b1e5d8fa8b8634",
  },
  Object {
    "document": "mutation ValidateOperations($gitContext:GitContextInput,$id:ID!,$operations:[OperationDocumentInput!]!,$tag:String){service(id:$id){__typename validateOperations(gitContext:$gitContext,operations:$operations,tag:$tag){__typename validationResults{__typename code description operation{__typename name}type}}}}",
    "metadata": Object {
      "engineSignature": "",
    },
    "signature": "600e261efe1f25ee30c53edd0b2b83c7eb6691ef5ae1c8f460ace2d303ad053d",
  },
  Object {
    "document": "fragment IntrospectionFullType on IntrospectionType{__typename description enumValues(includeDeprecated:true){__typename deprecationReason description isDeprecated name}fields{__typename args{__typename...IntrospectionInputValue}deprecationReason description isDeprecated name type{__typename...IntrospectionTypeRef}}inputFields{__typename...IntrospectionInputValue}interfaces{__typename...IntrospectionTypeRef}kind name possibleTypes{__typename...IntrospectionTypeRef}}fragment IntrospectionInputValue on IntrospectionInputValue{__typename defaultValue description name type{__typename...IntrospectionTypeRef}}fragment IntrospectionTypeRef on IntrospectionType{__typename kind name ofType{__typename kind name ofType{__typename kind name ofType{__typename kind name ofType{__typename kind name ofType{__typename kind name ofType{__typename kind name ofType{__typename kind name}}}}}}}}query GetSchemaByTag($tag:String!){service:me{__typename...on Service{schema(tag:$tag){__typename hash __schema:introspection{__typename directives{__typename args{__typename...IntrospectionInputValue}description locations name}mutationType{__typename name}queryType{__typename name}subscriptionType{__typename name}types{__typename...IntrospectionFullType}}}}}}",
    "metadata": Object {
      "engineSignature": "",
    },
    "signature": "b0832ede768852bf384d1067542249e4d936f9f158bea05e6262b67c15c2863c",
  },
]
`;
